### PR TITLE
[MM-10394] Make hidden town-square (on read-only) be available to view by joining in more channels

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Make hidden town-square (on read-only) be available to view by joining in more channels

Related PRs:
- https://github.com/mattermost/mattermost-server/pull/8751
- https://github.com/mattermost/mattermost-redux/pull/493
- https://github.com/mattermost/mattermost-webapp/pull/1194

Note: will update this PR once server and redux changes are merged.

#### Ticket Link
Jira ticket: [MM-10394](https://mattermost.atlassian.net/browse/MM-10394), also related to [MM-10430](https://mattermost.atlassian.net/browse/MM-10430)

#### Checklist
- [x] Added or updated unit tests (required for all new features)

#### Device Information
This PR was tested on: [iOS simulator and Android emulator] 

#### Screen captures
Android:
android_experimentalhide

iPhone:
ios_experimentalhide

### CODE REVIEW

**Comment by saturninoabril** [comment]
Thanks @enahum for the guidance!
I'll update this and also other PRs (redux, webapp) based from our discussion and the revised ticket.

**Comment by enahum** [comment]
@saturninoabril where are we on this one?

**Comment by saturninoabril** [comment]
I'll prioritize this (and other PRs) today and tomorrow.

File: app/screens/more_channels/more_channels.test.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import Adapter from 'enzyme-adapter-react-16';
+configure({adapter: new Adapter()});
+
+import MoreChannels from './more_channels.js';
+
+jest.mock('react-intl');
+
+describe('MoreChannels', () => {
+    const navigator = {
+        setOnNavigatorEvent: () => {}, // eslint-disable-line no-empty-function
```
[the remark this answers is not in the document]
Comment (saturninoabril) [reply app/screens/more_channels/more_channels.test.js]: re: ESLint rules, [maybe it's possible](https://github.com/eslint/eslint/blob/master/docs/user-guide/configuring.md#configuration-based-on-glob-patterns) but I'll give it a shot to investigate.
I agree with the use of `jest.fn()` only that recently, I'm avoiding it on baseProps. I usually use it whenever I have an immediate need to mock or spy a function.

**Comment by enahum** [comment]
@saturninoabril PR looks good, but I'll wait until the redux PR is merged and this PR updates the reference to the mattermost-redux repo.

sounds good?

**Pull request merged**